src/os/ogc/query/ogcquery.js

```diff
@@ -0,0 +1,126 @@
+goog.module('os.ogc.query');
+
+const query = goog.require('os.query');
+const ConfirmUI = goog.require('os.ui.window.ConfirmUI');
+const registry = goog.require('os.ogc.registry');
+const {directive: ogclistdirective} = goog.require('os.ui.ogc.List');
+
+
+/**
+ * Generic UI and handlers for an OGC Query
+ */
+class Query {
+  /**
+   * TYPE is required for service to work; override this and hard-code "super(type);" in the constructor
+   * @param {!string} type
+   */
+  constructor(type) {
+    /**
+     * @type {string}
+     */
+    this.type = type;
+
+    /**
+     * @type {string}
+     * @protected
+     */
+    this.eventType_ = 'ogcQuery:*';
+  }
+
+  /**
+   * Handle a area being chosen in the import area border dialog.
+   * @param {function(!ol.Feature, !ol.Feature)} callback The callback to fire when the area is loaded
+   * @param {Array<!ol.Feature>} datum The requested area
+   * @private
+   */
+  onChoice_(callback, datum) {
+    if (datum && datum.length > 0) {
+      const service = registry.getInstance().get(this.type);
+      if (service) {
+        // inject the requested area into the callback so it has some additional context
+        service
+            .get(datum[0])
+            .then(goog.partial(callback, datum));
```

Comment:
Nit: `goog.partial`'s functionality is replicated by calling `fn.bind(undefined, args)`.

Reply:
Replaced all the .partial()'s with .bind()'s in the OGC code.